### ISSUE TITLE
Enable linting etc.

### ISSUE THREAD

**Issue by espenfl** [aiida-vasp/parsevasp#31]
We should enable pre-commit checks and clean the code accordingly.

**Comment by atztogo** [comment]
I want to touch parservasp, but I am afraid of doing it because no pre-commit setting.

As an attempt, I used `.pre-commit-config.yaml` of aiida-vasp, and `pre-commit install && pre-commit run --all-files` changed significant amount of lines.

So I thin it is good to introduce pre-commit hook when almost nobody is working on parsevasp. Do you know when is appropriate. I assume only @espenfl is working if it is, but maybe also @zhubonan and @lan496?

**Comment by lan496** [comment]
I am not currently working on parsevasp. But, if possible, I would like the maintainers to merge [this open PR](https://github.com/aiida-vasp/parsevasp/pull/95) before introducing the linters (to avoid rebasing and re-opening the PR).

**Comment by espenfl** [comment]
> I am not currently working on parsevasp. But, if possible, I would like the maintainers to merge [this open PR](https://github.com/aiida-vasp/parsevasp/pull/95) before introducing the linters (to avoid rebasing and re-opening the PR).

Done. Thanks a lot for the contribution and sorry for the delay of merging.

**Comment by espenfl** [comment]
> good to introduce pre-commit hook when almost nobody is working on parsevasp.

Yes, absolutely. That was the plan, but I have never gotten to it. If you have the time, please look into it. If it is the same linter it should not really change anything substantial, even though there will be some changes of lines etc. Also we have a pretty decent test coverage.

**Comment by atztogo** [comment]
Thanks for your quick responses.
@espenfl, will you merge your PR #97?

I am going to send a PR of the linted version if it is ready to run pre-commit.

**Comment by espenfl** [comment]
Goodie, will do, but in the middle of fixing the merge conflicts for the `aiida-vasp` 527 PR. Soon done. But just open a PR and then I will update the stale PR for the sync of the eigenvalues based on that, no problem. Thanks @atztogo :)

**Comment by espenfl** [comment]
Yes, we should use the `setup.json` data from `aiida-core`. Since this is a dependency for both `aiida-vasp` and `parsevasp`. This is fine, but for `parsevasp` we need to also let it fly on its own so I suggest that we just inherit what is defined in `aiida-core` and only override if it is strictly necessary (do not really see any need at this point to pin something in `parsevasp`).

**Comment by atztogo** [comment]
Thanks for your valuable comments. I will follow `.pre-commit-config.yaml` in the latest release of aiida-core (v1.6.5). Do you agree with `max-line-length = 120` following aiida-core?

**Comment by espenfl** [comment]
> Thanks for your valuable comments. I will follow `.pre-commit-config.yaml` in the latest release of aiida-core (v1.6.5). Do you agree with `max-line-length = 120` following aiida-core?

Thanks yourself @atztogo Yes, I have no specific feelings for the max length, except that 80 is too short. Let us go with `120` as well and then we will just update accordingly if there will be changes.
